galite-core/src/main/kotlin/org/kopi/galite/chart/VChartType.kt

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2013-2020 kopiLeft Services SARL, Tunis TN
+ * Copyright (c) 1990-2020 kopiRight Managed Solutions GmbH, Wien AT
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1 as published by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package org.kopi.galite.chart
+
+import java.io.Serializable
+
+import org.kopi.galite.base.UComponent
+import org.kopi.galite.visual.VModel
+
+/**
+ * Supported chart types.
+ *
+ * @param ordinal The type ordinal.
+ * @param name The type name.
+ */
+open class VChartType protected constructor(val ordinal: Int, val name: String) : Serializable, CConstants, VModel {
+  // --------------------------------------------------------------------
+  // IMPLEMENTATIONS
+  // --------------------------------------------------------------------
+
+  override fun setDisplay(display: UComponent) {
+    assert(display is UChartType) { "The display should be a chart type view" }
+    this.display = display as UChartType
+  }
+
+  override fun getDisplay(): UComponent = display
+
+  /**
+   * Creates the data series objects from the chart model.
+   * @param chart The chart model.
+   */
+  fun createDataSeries(chart: VChart) {
+    dataSeries.clear()
```

Comment:
check if we should add this 
if (dataSeries == null) {
      dataSeries = LinkedList()
    }
in this case change type of dataSeries